### DISCUSSION TITLE
Best way to log in user with Nhost when logged into another app?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#3013] [Q&A] [answered]
We have an app that a user logs into with auth from that particular app (auth is handled by the app and we don't control that, but we know when the user is logged in). Once the user is logged in with that app, we want to log them in with Nhost, so that we can generate the necessary keys we can use for calls to the Hasura database. What is the best way to do this? I'd rather not have them log in separately on the app to Nhost, which was my first idea. I'd rather have something that says: isLoggedInApp is true, then run a query to Nhost and log them in with Nhost. I thought about doing an anonymous login, but then that would create alot of anonymous logins which will get confusing, as we want to know exactly who logged in. Anyway, what is the best way to  handle this type of situation? Thanks.

**Comment by dbarrosop** [comment]
Is the other app using oauth2 or openid connect?

**Comment by osseonews** [comment]
No. It has a custom auth application that is developed by the provider. Uses JWT tokens, but we are not in control of setting those tokens.

**Reply by dbarrosop** [reply]
Then it isn't possible. The only way to share identities is via openid connect or oauth2. The only thing you could do is share the JWT secret and have them add all the needed claims for hasura so the sessions are portable but make sure you understand the security implications. Another thing you could do is automatically sync users but that will depend on which capabilities the other provider has for that.

**Reply by ctrlemo** [reply]
https://docs.nhost.io/guides/auth/custom-jwts

I have this approach working with Auth0 on my project.

In my case Auth0 does allow the addition of custom claims to their JWT.

In your case you said it was not possible to add custom claims.

You could still follow the approach, by only verifying the JWT from your provider is valid within the nhost cloud ☁️ function.
And if it is, you create the nhost compatible version of the token including "hardcoded" custom claims, under the assumption that all users share the same role.
Then in Hasura you can permission based on the particular role to check if the incoming graphql query from a user is allowed to read write.

This is all assuming you are doing only permissions on role level.

If you need permission on actual user level, then like already @dbarrosop said, you will need some way to sync users id's, etc, from you provider over to a Nhost postgres table, that will allow you to join on it for permission checks.

**Reply by osseonews** [reply, marked as answer]
@ctrlemo Thanks! This is exactly the approach I was thinking about. Thanks for confirming that you did this as well and for providing more details.
